lib/AST/ASTPrinter.cpp

```diff
@@ -8315,8 +8315,39 @@ swift::getInheritedForPrinting(
         }
       }
     }
-
-    Results.push_back(inherited.getEntry(i));
+    
+    auto entry = inherited.getEntry(i);
+    // Desugar ParameterizedProtocolType from the entry
+    if (auto type = entry.getType()) {
+      Type desugaredType = type;
+      // Iterate over ProtocolCompositionType members
+      if (auto *compositionType = type->getAs<ProtocolCompositionType>()) {
+        SmallVector<Type, 4> desugaredMembers;
+        bool hasParameterized = false;
+        for (auto member: compositionType->getMembers()) {
+          if (auto *paramProtocolType = member->getAs<ParameterizedProtocolType>()) {
```

Comment:
The members may themselves be ProtocolCompositionTypes, so this actually needs to be recursive. You can factor out a helper for this purpose